google-cloud-storage/src/test/java/com/google/cloud/storage/conformance/retry/CtxFunctions.java

```diff
@@ -133,6 +141,34 @@ static final class Rpc {
         (ctx, c) -> ctx.map(state -> state.with(ctx.getStorage().create(state.getBlobInfo())));
   }
 
+  static final class Notification {
+    static final CtxFunction pubsubTopic =
+        (ctx, c) -> {
+          String projectId = c.getProjectId();
+          TopicName name = TopicName.of(projectId, c.getTopicName());
+          return ctx.map(s -> s.with(name));
+        };
+
+    static final CtxFunction notification =
+        (ctx, c) -> ctx.map(state -> {
+          PayloadFormat format = PayloadFormat.JSON_API_V1;
+          Map<String, String> attributes = ImmutableMap.of("label1", "value1");
+          NotificationInfo notificationInfo =
+              NotificationInfo.newBuilder(state.getTopic().toString())
+                  .setCustomAttributes(attributes)
+                  .setPayloadFormat(format)
+                  .build();
+          return state.with(
+              ctx.getStorage().createNotification(c.getBucketName(), notificationInfo));
+        });
+
+    static final CtxFunction pubsubTopicSetup =
+        defaultSetup.andThen(pubsubTopic);
+
+    static final CtxFunction notificationSetup =
+        pubsubTopicSetup.andThen(notification);
```

Comment:
I think these new CtxFunctions can simply be added to the existing `ResourceSetup` class to avoid the need for additional imports/qualified references.